### PR TITLE
feat: add Swift codes for array article

### PR DESCRIPTION
**Pull request by nuomi1**
If this PR is related to coding or code translation, please fill out the checklist.

- [x] I've tested the code and ensured the outputs are the same as the outputs of reference codes.
- [x] I've checked the codes (formatting, comments, indentation, file header, etc) carefully.
- [x] The code does not rely on a particular environment or IDE and can be executed on a standard system (Win, macOS, Ubuntu).

### CODE REVIEW

**Comment by nuomi1** [comment]
```text
result of `array.swift`

数组 arr = [0, 0, 0, 0, 0]
数组 nums = [1, 2, 3, 4, 5]
在 nums 中获取随机元素 1
将数组长度扩展至 8 ，得到 nums = [1, 2, 3, 4, 5, 0, 0, 0]
在索引 3 处插入数字 6 ，得到 nums = [1, 2, 3, 6, 4, 5, 0, 0]
删除索引 2 处的元素，得到 nums = [1, 2, 6, 4, 5, 0, 0, 0]
在 nums 中查找元素 3 ，得到索引 = -1
```

**Comment by krahets** [comment]
Please modify the test case for consistency

```
数组 arr = [0, 0, 0, 0, 0]
数组 nums = [1, 3, 2, 5, 4]
在 nums 中获取随机元素 4
将数组长度扩展至 8 ，得到 nums = [1, 3, 2, 5, 4, 0, 0, 0]
在索引 3 处插入数字 6 ，得到 nums = [1, 3, 2, 6, 5, 4, 0, 0]
删除索引 2 处的元素，得到 nums = [1, 3, 6, 5, 4, 0, 0, 0]
在 nums 中查找元素 3 ，得到索引 = 1
```

File: codes/swift/chapter_array_and_linkedlist/array.swift [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+}
+
+@main
+enum _Array {
+    // Driver Code
+    static func main() {
+        // 初始化数组
+        let arr = Array(repeating: 0, count: 5)
+        print("数组 arr = \(arr)")
+        var nums = [1, 2, 3, 4, 5]
```
Comment (krahets) [inline codes/swift/chapter_array_and_linkedlist/array.swift:77]: [1 ,3 ,2 ,5, 4]

File: docs/chapter_array_and_linkedlist/array.md [hunk 2]
```diff
@@ -81,6 +81,14 @@ comments: true
     int[] nums = { 1, 3, 2, 5, 4 };
     ```
 
+=== "Swift"
+
+    ```swift title="array.swift"
+    // 初始化数组
+    let arr = Array(repeating: 0, count: 5) // [0, 0, 0, 0, 0]
+    let nums = [1, 2, 3, 4, 5]
```
Comment (krahets) [inline docs/chapter_array_and_linkedlist/array.md:89]: [1, 3, 2, 5, 4]

**Review by krahets — COMMENTED** [review]
Thanks! Please address the comments.

File: codes/swift/chapter_array_and_linkedlist/array.swift (hunk 1 above)
Comment (nuomi1) [reply codes/swift/chapter_array_and_linkedlist/array.swift]: Fixed.

File: docs/chapter_array_and_linkedlist/array.md (hunk 2 above)
Comment (nuomi1) [reply docs/chapter_array_and_linkedlist/array.md]: Fixed.

**Review by krahets — APPROVED** [review]
Now merge it.

**Pull request merged**